components/support/utils/src/main/java/mozilla/components/support/utils/WebURLFinder.kt

```diff
@@ -75,7 +75,7 @@ class WebURLFinder {
     companion object {
         // Taken from mozilla.components.support.utils.URLStringUtils. See documentation
         // there for a complete description.
-        private const val autolinkWebUrlPattern = "(\\w+-)*\\w+(://[/]*|:|\\.)(\\w+-)*\\w+([\\S&&[^\\w-]]\\S*)?"
+        private const val autolinkWebUrlPattern = "^\\s*(\\w+-+)*\\w+(://[/]*|:|\\.)(\\w+-+)*\\w+([\\S&&[^\\w-]]\\S*)?\\s*\$"
```

Comment:
By changing this, afaict, we're changing the behavior of all consumers of WebUrlFinder (which are 2 instances of clipboard code and `TabIntentProcessor`). While these bits of code will also have performance issues, I'm concerned that the requirements for those may not be as lenient as the requirements for the clipboard stuff. Figuring this out seems like it's outside the scope of this change.

**Can we change this behavior _just_ for the clipboard code that has performance issues?** We should probably use `isURLLenient` rather than duplicating the logic.